packages/react/src/primitives/Table/__tests__/Table.test.tsx

```diff
@@ -0,0 +1,87 @@
+import { render, screen } from '@testing-library/react';
+
+import { Table } from '../Table';
+import { ComponentClassNames } from '../../shared';
+import { TableProps } from '../../types';
+
+describe('Table primitive', () => {
+  const testCaption = 'test-caption';
+  const testSummary = 'test-summary';
+
+  const setup = async (props: TableProps = {}) => {
+    render(<Table {...props} />);
+
+    return {
+      $table: await screen.findByRole('table'),
+    };
+  };
+
+  describe('Accessibility', () => {
+    it('should render a table with an accessible label', async () => {
+      const label = 'accessible-label';
+      const { $table } = await setup({ label });
+
+      expect($table).toHaveAttribute('aria-label', label);
```

Comment:
nit: Maybe you can try to use this matcher for accessible description specifically? `expect().toHaveAccessibleDescription()`

Reply:
Nice callout, I actually learned a decent amount about accessible tables. Best practice is to use the `caption` tag which is used by screen readers to convey table information. You can read some more about it here: https://www.w3.org/WAI/tutorials/tables/caption-summary/

I updated our unit tests to test the actual caption by doing a query for the `table` role using the caption provided.